plugins/outputs/inlong/inlong.go

```diff
@@ -0,0 +1,107 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package inlong
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/apache/inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-golang/dataproxy"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Inlong struct {
+	ManagerURL string          `toml:"url"`
+	GroupID    string          `toml:"group_id"`
+	StreamID   string          `toml:"stream_id"`
+	Log        telegraf.Logger `toml:"-"`
+
+	producer   dataproxy.Client
+	serializer telegraf.Serializer
+}
+
+func (*Inlong) SampleConfig() string {
+	return sampleConfig
+}
+
+func (i *Inlong) Init() error {
+	if i.ManagerURL == "" {
+		return errors.New("'url' must not be empty")
+	}
+	if i.GroupID == "" {
+		return errors.New("''group_id' must not be empty")
+	}
+	if i.StreamID == "" {
+		return errors.New("'stream_id' must not be empty")
+	}
+	parsedURL, err := url.Parse(i.ManagerURL)
+	if err != nil {
+		return fmt.Errorf("invalid URL: %s", i.ManagerURL)
+	}
+
+	switch parsedURL.Scheme {
+	case "http", "https":
+		if parsedURL.Host == "" {
+			return fmt.Errorf("no host in URL: %s", i.ManagerURL)
+		}
+	default:
+		return fmt.Errorf("invalid URL scheme: %s", parsedURL.Scheme)
+	}
+	elements := []string{"inlong", "manager", "openapi", "dataproxy", "getIpList"}
+	i.ManagerURL = parsedURL.JoinPath(elements...).String()
+	return nil
+}
+
+func (i *Inlong) SetSerializer(serializer telegraf.Serializer) {
+	i.serializer = serializer
+}
+
+func (i *Inlong) Connect() error {
+	producer, err := dataproxy.NewClient(
+		dataproxy.WithGroupID(i.GroupID),
+		dataproxy.WithURL(i.ManagerURL),
+	)
+	if err != nil {
+		i.Log.Errorf("Connect to manager %v group_id %v error %v", i.ManagerURL, i.GroupID, err)
+		return &internal.StartupError{Err: err, Retry: true}
+	}
+	i.producer = producer
+	return nil
+}
+
+func (i *Inlong) Close() error {
+	i.producer.Close()
+	return nil
+}
+
+func (i *Inlong) Write(metrics []telegraf.Metric) error {
+	for _, metric := range metrics {
+		b, err := i.serializer.Serialize(metric)
+		if err != nil {
+			return fmt.Errorf("could not serialize metric: %w", err)
+		}
+		err = i.producer.Send(context.Background(), dataproxy.Message{
+			GroupID:  i.GroupID,
+			StreamID: i.StreamID,
+			Payload:  b,
+		})
+		if err != nil {
+			return fmt.Errorf("could not send metric to GroupID %s StreamID %s: %w", i.GroupID, i.StreamID, err)
+		}
+	}
```

Comment:
Do we really need to send this metric-by-metric? Can't we send all metrics together?

Reply:
> Do we really need to send this metric-by-metric? Can't we send all metrics together?

The SDK will assemble and send in batches internally

Reply:
Yeah what I meant is, you are currently serializing each metric individually, but you also could serialize the whole set of metrics and send it with one call like
```suggestion
	b, err := i.serializer.SerializeBatch(metrics)
	if err != nil {
		return fmt.Errorf("could not serialize metrics: %w", err)
	}
	err = i.producer.Send(context.Background(), dataproxy.Message{
		GroupID:  i.GroupID,
		StreamID: i.StreamID,
		Payload:  b,
	})
	if err != nil {
		return fmt.Errorf("could not send metric to GroupID %s StreamID %s: %w", i.GroupID, i.StreamID, err)
	}
```

Reply:
> Yeah what I meant is, you are currently serializing each metric individually, but you also could serialize the whole set of metrics and send it with one call like
> 
> I'm not sure if that make sense to consumers of the message, that's why I'm asking. If both variants make sense, we can implement it later in a separate PR.

This will cause the inlong platform to recognize these multiple messages as one, and I mistakenly merged them into it. I will revert again